src/object/patch/add-link.js

```diff
@@ -3,181 +3,81 @@
-            node2 = new DAGNode(Buffer.from('some other node'))
-          } catch (err) {
-            return cb(err)
-          }
-
-          cb()
-        },
-        (cb) => {
-          // note: we need to put the linked obj, otherwise IPFS won't
-          // timeout. Reason: it needs the node to get its size
-          ipfs.object.put(node2, (err, cid) => {
-            expect(err).to.not.exist()
-
-            cb()
-          })
-        },
-        (cb) => {
-          asDAGLink(node2, 'link-to-node', (err, link) => {
-            expect(err).to.not.exist()
-
-            node1b = new DAGNode(node1a.Data, node1a.Links.concat(link))
-
-            cb()
-          })
-        },
-        (cb) => {
-          ipfs.object.put(node1b, (err, cid) => {
-            expect(err).to.not.exist()
-
-            node1bCid = cid
-
-            cb()
-          })
-        },
-        (cb) => {
-          ipfs.object.patch.addLink(testNodeCid, node1b.Links[0], (err, cid) => {
-            expect(err).to.not.exist()
-            expect(node1bCid).to.eql(cid)
-            cb()
-          })
-        }
-        /* TODO: revisit this assertions.
-        (cb) => {
-          // note: make sure we can link js plain objects
-          const content = Buffer.from(JSON.stringify({
-            title: 'serialized object'
-          }, null, 0))
-          ipfs.add(content, (err, result) => {
-            expect(err).to.not.exist()
-            expect(result).to.exist()
-            expect(result).to.have.lengthOf(1)
-            const object = result.pop()
-            node3 = {
-              name: object.hash,
-              multihash: object.hash,
-              size: object.size
-            }
-            cb()
-          })
-        },
-        (cb) => {
-          ipfs.object.patch.addLink(testNodeWithLinkMultihash, node3, (err, node) => {
-            expect(err).to.not.exist()
-            expect(node).to.exist()
-            testNodeWithLinkMultihash = node.multihash
-            testLinkPlainObject = node3
-            cb()
-          })
-        }
-        */
-      ], done)
-    })
+      const testNodeCid = await ipfs.object.put(obj)
+      const node1a = new DAGNode(obj.Data, obj.Links)
+      const node2 = new DAGNode(Buffer.from('some other node'))
 
-    it('should add a link to an existing node (promised)', async () => {
-      const obj = {
-        Data: Buffer.from('patch test object (promised)'),
-        Links: []
+      // note: we need to put the linked obj, otherwise IPFS won't
+      // timeout. Reason: it needs the node to get its size
+      await ipfs.object.put(node2)
+
+      const link = await asDAGLink(node2, 'link-to-node')
+      const node1b = new DAGNode(node1a.Data, node1a.Links.concat(link))
+
+      const node1bCid = await ipfs.object.put(node1b)
+
+      const cid = await ipfs.object.patch.addLink(testNodeCid, node1b.Links[0])
+      expect(node1bCid).to.eql(cid)
+
+      /* TODO: revisit this assertions.
```

Comment:
Can this be uncommented?

Reply:
I'm having a hard time trying to figure several things out:
```js
/* TODO: revisit this assertions.
      // note: make sure we can link js plain objects
      const content = Buffer.from(JSON.stringify({
        title: 'serialized object'
      }, null, 0))
      const result = await ipfs.add(content)
      expect(result).to.exist()
      expect(result).to.have.lengthOf(1)
      const object = result.pop()
      const node3 = {
        name: object.hash,
        multihash: object.hash,
        size: object.size
      }
      const node = await ipfs.object.patch.addLink(testNodeWithLinkMultihash, node3)
      expect(node).to.exist()
      testNodeWithLinkMultihash = node.multihash
      testLinkPlainObject = node3
      */
```

This seems to be a mix between code and pseudo-code. I don't know if I did correctly understand the goal of this test but I did try to do this: `const node = await ipfs.object.patch.addLink(node1bCid, node3)` where `node1bCid` is the cid of the node with link (as `testNodeWithLinkMultihash` indicates). However, I'm getting this error: `Error: invalid path "": path does not begin with '/'`.

Furthermore, this seems to be a different test (like a second test, testing plain objects this time) and thus, I think it should have its own test, separately of `should add a link to an existing node` test. 

So, we can leave it as it currently is (commented), delete this test, or if you really want to keep it can you please help me? @achingbrain